### PR TITLE
website(consul-api-gateway): add ReferencePolicy to overview docs

### PR DESCRIPTION
**Pull request by mikemorris**
### Description

Add brief documentation of ReferencePolicy for cross-namespace BackendRefs in HTTPRoute and TCPRoute to Consul API Gateway docs.

### TODO
- [x] Add Kubernetes object to [Overview](http://consul.io/docs/api-gateway) page
- [x] Bump HashiCorp Consul Helm chart version in [Technical Specs](http://consul.io/docs/api-gateway/tech-specs) section of install docs
  - https://github.com/hashicorp/consul-k8s/releases/tag/v0.43.0
  - https://github.com/hashicorp/consul-k8s/pull/1148
  - https://github.com/hashicorp/consul-k8s/pull/1169
- [x] Add note to [`Route`](http://consul.io/docs/api-gateway/consul-api-gateway-install#route) section of install docs

### Links

https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferencePolicy

### PR Checklist

* [ ] ~~updated test coverage~~
* [x] external facing docs updated
* [ ] ~~not a security concern~~
* [ ] ~~checklist [folder](./../docs/config) consulted~~

### CODE REVIEW

File: website/content/docs/api-gateway/consul-api-gateway-install.mdx [hunk 1]
```diff
@@ -305,6 +305,49 @@ spec:
 
 </CodeBlockConfig>
 
+If you wish to create a route in a namespace other than the namespace the backendRef is in, you will need to additionally
```
Comment (mikemorris) [inline website/content/docs/api-gateway/consul-api-gateway-install.mdx:308]:
```suggestion
If you wish to create a route for a backendRef in a different namespace, you will need to additionally
```
Suggested rephrasing to refocus from the perspective of the route rather than from the backendRef.

I see three possible deployment patterns:
- gateway in separate namespace managed by operator, dev teams manage routes for their own services (requires setting more permissive `allowedRoutes` config on gateway)
- gateway and routes managed by operator in same namespace, pointing to dev team services in other namespaces (requires reference policies)
- routes in a different namespace than both the gateway and services (this is just weird)

Comment (mikemorris) [inline website/content/docs/api-gateway/consul-api-gateway-install.mdx:308]:
```suggestion
If you wish to create a route for a backendRef in a different namespace, you will need to additionally
```
Suggested rephrasing to refocus from the perspective of the route rather than from the backendRef.

I see three possible deployment patterns:
- gateway in separate namespace managed by operator, dev teams manage routes for their own services (requires setting more permissive `allowedRoutes` config on gateway, no reference policy needed for most cases)
- gateway and routes managed by operator in same namespace, pointing to dev team services in other namespaces (requires reference policies)
- routes in a different namespace than both the gateway and services (this is just weird)

File: website/content/docs/api-gateway/consul-api-gateway-install.mdx [hunk 2]
```diff
@@ -305,6 +305,49 @@ spec:
...
+create a [ReferencePolicy](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferencePolicy).
+
+The following example creates a route named `example-route` in namespace `route-namespace` with a backendRef in namespace `service-namespace` Traffic is allowed because of the ReferencePolicy named `reference-policy` in namespace `service-namespace` that allows traffic from HTTPRoutes in `route-namespace` to Services in `service-namespace`.
```
Comment (mikemorris) [inline website/content/docs/api-gateway/consul-api-gateway-install.mdx:311]:
```suggestion
The following example creates a route named `example-route` in namespace `gateway-namespace` with a backendRef in namespace `service-namespace` Traffic is allowed because of the ReferencePolicy named `reference-policy` in namespace `service-namespace` that allows traffic from HTTPRoutes in `gateway-namespace` to Services in `service-namespace`.
```

File: website/content/docs/api-gateway/consul-api-gateway-install.mdx [hunk 3]
```diff
@@ -305,6 +305,49 @@ spec:
...
+
+<CodeBlockConfig filename="route_with_referencepolicy.yaml">
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: example-route
+  namespace: route-namespace
```
Comment (mikemorris) [inline website/content/docs/api-gateway/consul-api-gateway-install.mdx:320]:
```suggestion
  namespace: gateway-namespace
```

File: website/content/docs/api-gateway/consul-api-gateway-install.mdx [hunk 4]
```diff
@@ -305,6 +305,49 @@ spec:
...
+spec:
+  parentRefs:
+    - name: example-gateway
+      namespace: gateway-namespace
```
Comment (mikemorris) [inline website/content/docs/api-gateway/consul-api-gateway-install.mdx:324]:
```suggestion
```
Targeting a gateway in a different namespace from the route would require setting a more permissive [`allowedRoutes`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1alpha2.AllowedRoutes) config on the gateway which we don't describe here.

File: website/content/docs/api-gateway/consul-api-gateway-install.mdx [hunk 5]
```diff
@@ -305,6 +305,49 @@ spec:
...
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: ReferencePolicy
+metadata:
+  name: reference-policy
+  namespace: service-namespace
+spec:
+  from:
+    - group: gateway.networking.k8s.io
+      kind: HTTPRoute
+      namespace: route-namespace
```
Comment (mikemorris) [inline website/content/docs/api-gateway/consul-api-gateway-install.mdx:342]:
```suggestion
      namespace: gateway-namespace
```

**Review by mikemorris — COMMENTED** [review]
Suggested a minor shift to put the HTTPRoute in `gateway-namespace` for the install guide example (to match what I think will be a more common deployment pattern), but overall LGTM!

**Review by nathancoleman — CHANGES_REQUESTED** [review]

File: website/content/docs/api-gateway/consul-api-gateway-install.mdx [hunk 6]
```diff
@@ -305,6 +305,47 @@ spec:
 
 </CodeBlockConfig>
 
+If you wish to create a route for a `backendRef` in a different namespace, you will need to additionally
+create a [ReferencePolicy](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferencePolicy).
+
+The following example creates a route named `example-route` in namespace `gateway-namespace` with a `backendRef` in namespace `service-namespace` Traffic is allowed because of the `ReferencePolicy` named `reference-policy` in namespace `service-namespace` that allows traffic from `HTTPRoutes` in `gateway-namespace` to `Services` in `service-namespace`.
```
Comment (nathancoleman) [inline website/content/docs/api-gateway/consul-api-gateway-install.mdx:311]:
```suggestion
The following example creates a route named `example-route` in namespace `gateway-namespace` with a `backendRef` in namespace `service-namespace`. Traffic is allowed because of the `ReferencePolicy` named `reference-policy` in namespace `service-namespace` which allows traffic from `HTTPRoutes` in `gateway-namespace` to `Services` in `service-namespace`.
```

File: website/data/docs-nav-data.json [hunk 7]
```diff
@@ -382,8 +382,8 @@
         "path": "api-gateway"
       },
       {
-        "title": "Usage",
-        "path": "api-gateway/api-gateway-usage"
```
Comment (nathancoleman) [inline website/data/docs-nav-data.json:386]: Is there a way to alias this so we don't break any outstanding links?

Comment (mikemorris) [reply website/data/docs-nav-data.json:386]: I believe this is an artifact of the base branch being rebased and a redirect was already added when this change was made.

Comment (mikemorris) [reply website/data/docs-nav-data.json]: See https://github.com/hashicorp/consul/pull/12811 for how the redirect was added.

File: website/content/docs/api-gateway/consul-api-gateway-install.mdx [hunk 8]
```diff
@@ -305,6 +305,48 @@ spec:
...
```
Comment (boruszak) [inline website/content/docs/api-gateway/consul-api-gateway-install.mdx]:
```suggestion
The following example creates a route named `example-route` in namespace `gateway-namespace`. This route has a `backendRef` in namespace `service-namespace`. Traffic is allowed because the `ReferencePolicy`, named `reference-policy` in namespace `service-namespace`, allows traffic from `HTTPRoutes` in `gateway-namespace` to `Services` in `service-namespace`.
```

Breaking this single sentences into multiple sentences to make it easier to understand the relationships betweent he similarly named routes and namespaces.

Comment (nathancoleman) [reply website/content/docs/api-gateway/consul-api-gateway-install.mdx]: I think the suggestion is failing to apply because it was based on the state of things prior to https://github.com/hashicorp/consul/pull/12861/commits/b22db7a1155025ae3200da94ee2d179ac9ed1974

File: website/content/docs/api-gateway/consul-api-gateway-install.mdx [hunk 9]
```diff
@@ -305,6 +305,48 @@ spec:
 
 </CodeBlockConfig>
 
+To create a route for a `backendRef` in a different namespace, you must also
+create a [ReferencePolicy](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferencePolicy).
+
+The following example creates a route named `example-route` in namespace `gateway-namespace` with a `backendRef` in namespace `service-namespace`. Traffic is allowed because of the `ReferencePolicy` named `reference-policy` in namespace `service-namespace` which allows traffic from `HTTPRoutes` in `gateway-namespace` to `Services` in `service-namespace`.
```
Comment (mikemorris) [inline website/content/docs/api-gateway/consul-api-gateway-install.mdx:311]:
```suggestion
The following example creates a route named `example-route` in namespace `gateway-namespace`. This route has a `backendRef` in namespace `service-namespace`. Traffic is allowed because of the `ReferencePolicy`, named `reference-policy` in namespace `service-namespace`, allows traffic from `HTTPRoutes` in `gateway-namespace` to `Services` in `service-namespace`.
```

File: website/content/docs/api-gateway/consul-api-gateway-install.mdx (hunk 8 above)
Comment (mikemorris) [reply website/content/docs/api-gateway/consul-api-gateway-install.mdx]: Redid this change in a new commit below.

File: website/content/docs/api-gateway/consul-api-gateway-install.mdx (hunk 9 above)
Comment (nathancoleman) [reply website/content/docs/api-gateway/consul-api-gateway-install.mdx]: @mikemorris there's a stray "of" that makes it read incorrectly
> allowed because of the `ReferencePolicy`, ...

File: website/content/docs/api-gateway/consul-api-gateway-install.mdx [hunk 10]
```diff
@@ -305,6 +305,48 @@ spec:
 
 </CodeBlockConfig>
 
+To create a route for a `backendRef` in a different namespace, you must also
+create a [ReferencePolicy](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferencePolicy).
+
+The following example creates a route named `example-route` in namespace `gateway-namespace`. This route has a `backendRef` in namespace `service-namespace`. Traffic is allowed because of the `ReferencePolicy`, named `reference-policy` in namespace `service-namespace`, allows traffic from `HTTPRoutes` in `gateway-namespace` to `Services` in `service-namespace`.
```
Comment (mikemorris) [inline website/content/docs/api-gateway/consul-api-gateway-install.mdx:311]:
```suggestion
The following example creates a route named `example-route` in namespace `gateway-namespace`. This route has a `backendRef` in namespace `service-namespace`. Traffic is allowed because the `ReferencePolicy`, named `reference-policy` in namespace `service-namespace`, allows traffic from `HTTPRoutes` in `gateway-namespace` to `Services` in `service-namespace`.
```

Comment (mikemorris) [reply website/content/docs/api-gateway/consul-api-gateway-install.mdx:311]: cc @nathancoleman

**Review by boruszak — APPROVED** [review]
Suggestions for sentence-level changes to improve clarity.

File: website/content/docs/api-gateway/index.mdx [hunk 11]
```diff
@@ -24,7 +24,7 @@ Consul API Gateway can be deployed on Kubernetes-based runtime environments and
 
 API Gateway routes traffic to services connected to the same service mesh. Those services can be running on the same Kubernetes cluster as the API Gateway, a different Kubernetes cluster, or another runtime, as long as they are connected to the same service mesh deployment and reachable over the network.
 
-Consul API Gateway implements, and is configured through, the Kubernetes [Gateway API Specification](https://gateway-api.sigs.k8s.io/). This specification defines a set of custom resource definitions (CRD) that can create logical gateways and routes based on the path or protocol of a client request. 
+Consul API Gateway implements, and is configured through, the Kubernetes [Gateway API Specification](https://gateway-api.sigs.k8s.io/). This specification defines a set of custom resource definitions (CRD) that can create logical gateways and routes based on the path or protocol of a client request.
```
[the remark this answers is not in the document]
Comment (mikemorris) [reply website/content/docs/api-gateway/index.mdx:27]: cc @Jeff-Apple for review of this phrasing change

File: website/content/docs/api-gateway/index.mdx [hunk 12]
```diff
@@ -38,8 +38,8 @@ are used, see the [documentation in our GitHub repo](https://github.com/hashicor
...
 | [`TCPRoute`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.TCPRoute) | <ul><li>Supported backend types: <ol><li>`core/v1/Service` (must map to a registered Consul service)</li><li>`api-gateway.consul.hashicorp.com/v1alpha1/MeshService`</li></ol></li></ul> |
+| [`ReferencePolicy`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferencePolicy) | <ul><li>Required to allow any reference from an `HTTPRoute` or `TCPRoute` to a Kubernetes `core/v1/Service` in a different namespace.</li><li>A route with an unpermitted BackendRef caused by the lack of a ReferencePolicy sets a `ResolvedRefs` status to `False` with the reason `RefNotPermitted`. The gateway listener rejects routes with an unpermitted BackendRef.</li><li>WARNING: If a route BackendRef becomes unpermitted, the entire route is removed from the gateway listener. <ul><li>BackendRef can lose permissions when you delete a Reference Policy or add a new unpermitted BackendRef to an existing route.</li></ul></li></ul> |
```
Comment (mikemorris) [inline website/content/docs/api-gateway/index.mdx:41]:
```suggestion
| [`ReferencePolicy`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferencePolicy) | <ul><li>Required to allow any reference from an `HTTPRoute` or `TCPRoute` to a Kubernetes `core/v1/Service` in a different namespace.</li><li>A route with an unpermitted BackendRef caused by the lack of a ReferencePolicy sets a `ResolvedRefs` status to `False` with the reason `RefNotPermitted`. The gateway listener rejects routes with an unpermitted BackendRef.</li><li>WARNING: If a route BackendRef becomes unpermitted, the entire route is removed from the gateway listener. <ul><li>A BackendRef can become unpermitted when you delete a Reference Policy or add a new unpermitted BackendRef to an existing route.</li></ul></li></ul> |
```

File: website/content/docs/api-gateway/index.mdx (hunk 11 above)
[the remark this answers is not in the document]
Comment (Jeff-Apple) [reply website/content/docs/api-gateway/index.mdx:27]:
```suggestion
Consul API Gateway implements and is configured through the Kubernetes [Gateway API Specification](https://gateway-api.sigs.k8s.io/). This specification defines a set of custom resource definitions (CRD) that can create logical gateways, and routes traffic based on a client request's path or protocol.
```
In the last sentence I didn't have a comma after gateway (or should it be after the "and" after gateway?) and I didn't have "traffic" in there.  This is now correct in its intended meaning.

[the remark this answers is not in the document]
Comment (boruszak) [reply website/content/docs/api-gateway/index.mdx:27]: One minor comment/question on the comma after "gateways": what routes the traffic, the specification or the CRD?

**If the specification routes, it should be:** "This specification defines a set of custom resource definitions (CRD) that can create logical gateways and routes traffic based on a client request's path or protocol.

**If the CRD routes, it should be:** "This specification defines a set of custom resource definitions (CRD) that can create logical gateways and route traffic based on a client request's path or protocol."

In short, the comma isn't strictly necessary for conjoining dependent clauses. It seems like the comma is being inserted to separate the "create gateways" and "routes traffic" verb/direct objects, but the difference in noun/verb case agreement ("specification...defines" vs. "definitions can create") already does that heavy lifting. As a result, the comma introduces uncertainty into the sentence.

[the remark this answers is not in the document]
Comment (mikemorris) [reply website/content/docs/api-gateway/index.mdx:27]: CRD would be doing the routing, latter sentence reads as expected to me.
